src/app/auth/services/auth.service.ts

```diff
@@ -410,9 +420,101 @@ export const unblockAccount = (token: string): Promise<void> => {
   return authClient.unblockAccount(token);
 };
 
+export const signUp = async (
```

Comment:
Instead of passing multiple parameters, you can pass a single object that contains all the necessary properties:

```
type SignUpParams = {
  doSignUp: RegisterFunction | UpdateInfoFunction;
  email: string;
  password: string;
  token: string;
  isNewUser: boolean;
  redeemCodeObject: boolean;
  dispatch: AppDispatch;
};

export const signUp = async (params: SignUpParams) => {
  const { doSignUp, email, password, token, isNewUser, redeemCodeObject, dispatch } = params;
  // Function implementation
}
```